lib/src/test/kotlin/app/cash/kfsm/InvalidStateTransitionTest.kt

```diff
@@ -13,4 +14,12 @@ class InvalidStateTransitionTest : StringSpec({
     InvalidStateTransition(LetterTransition(States(C, B), D), Letter(E)).message shouldBe
       "Value cannot transition {B, C} to D, because it is currently E. [value=Letter(state=E)]"
   }
+
+  "exposes transition and state" {
+    val error = InvalidStateTransition(LetterTransition(A, B), Letter(E))
+
+    error.transition.from.set shouldContain A
+    error.transition.to shouldBe B
+    error.value.state shouldBe E
```

Comment:
This must be `Any`?

Reply:
It is worth just copying the data class into a client of kfsm like sparky or moongate and making sure `error.value.state` is useful in, say, a where clause?

Reply:
or more easily, try a where clause right here instead of the equality assertion

Reply:
@Synesso when you say where clause, do you mean filtering for a specific state?

Reply:
I've tested in moongate, working as expected with `reified` to capture the type, avoiding type erasure.

Exposing the transition also works, but interactions are a bit more cumbersome:
```kt
inline fun <reified T : Transition<V, S>, S : State<S>, V : Value<V, S>> getCurrentTransition(): Transition<V, S>? =
    transition as? T

// Usage
InvalidStateTransition(...).getCurrentTransition<LetterTransition, Char, Letter>()
```

I think `getState` is more useful.